components/resources/library/src/commonTest/kotlin/org/jetbrains/compose/resources/TestUtils.kt

```diff
@@ -66,4 +74,49 @@ internal fun parsePluralSamples(samples: String): List<Int> {
 internal fun pluralRuleListOf(vararg rules: Pair<PluralCategory, String>): PluralRuleList {
     val pluralRules = rules.map { PluralRule(it.first, it.second) } + PluralRule(PluralCategory.OTHER, "")
     return PluralRuleList(pluralRules.toTypedArray())
-}
\ No newline at end of file
+}
+
+/**
+ * Executes a test block within a Compose UI testing environment while ensuring
+ * that any cached resources are cleared before the test begins.
+ */
+@OptIn(ExperimentalTestApi::class)
+fun runComposeResourceTest(block: ComposeUiTest.() -> Unit) = runComposeUiTest {
+    var areCachesCleared = false
+    setContent {
+        LaunchedEffect(Unit) {
+            ResourceCaches.asyncClear()
+            areCachesCleared = true
+        }
+    }
+    waitUntil { areCachesCleared }
+    block()
+}
+
+/**
+ * Injects resource cache clearing to the content and returns a callback to clear resource caches.
+ * Used for tests that need to test some conditions after cache clearing.
+ * Note: a returned call-back can be used only once.
+ */
+@OptIn(ExperimentalTestApi::class)
+fun ComposeUiTest.setContentWithResourceCacheCleaning(content: @Composable () -> Unit): ComposeUiTest.()->Unit {
+    var clearCaches by mutableStateOf(false)
+    var areCachesCleared = false
+
+    this.setContent {
+        content()
+        LaunchedEffect(clearCaches) {
+            if (clearCaches) {
+                ResourceCaches.asyncClear()
+                areCachesCleared = true
+            }
+        }
+    }
+
+    return {
+        clearCaches = true
+        waitUntil { areCachesCleared }
+    }
```

Comment:
as we briefly discussed during the call, this indeed can be changed too.

we can have a function:

```
@OptIn(ExperimentalTestApi::class)
fun ComposeUiTest.cleanResourceCache() {
    var areCachesCleared = false
    
    MainScope().launch(start = CoroutineStart.UNDISPATCHED) {
        ResourceCaches.asyncClear()
        areCachesCleared = true
    }
    
    waitUntil { areCachesCleared }
}
```

A test can call it whener it needs. And there is no need for an additional `setContentWithResourceCacheCleaning` - `setContent` will be enough.